### ISSUE TITLE
Memory Leaks in SubViewport When Window is Minimized

### ISSUE THREAD

**Issue by MizunagiKB** [godotengine/godot#90030]
### Tested versions

4.2.stable

### System information

Godot v4.2.stable - macOS 14.3.1 - Vulkan (Forward+) - integrated Apple M2 Max - Apple M2 Max (12 Threads)

### Issue description

I've observed that memory leaks occur when explicit initialization (setting to transparent_bg) is performed every time when using SubViewport.
This issue could lead to significant performance issues and needs to be resolved.

> Additional Information:
This issue is not specifically related to whether the SubViewport is connected to a node.
It seems to occur regardless of the node connection status of the SubViewport.

This can be reproduced with the following code:

```gdscript
extends Node2D

var v: SubViewport

func _ready():
    v = SubViewport.new()
    
func _process(_delta):
    v.transparent_bg = true
```

![mleak_vp_transparent_bg](https://github.com/godotengine/godot/assets/5047878/b8339416-48fd-4e05-92d6-0b28ad0f8b87)

> This issue was discovered during the investigation of the following issue:
> https://github.com/godotengine/godot/issues/89651
> https://github.com/godotengine/godot/issues/90017

### Steps to reproduce

To reproduce this issue, execute the mleak_vp_transparent_bg.tscn file included in the attached project.

While the application is running, open the Debugger/Monitors in the Editor and pay attention to the value of Memory/Static.

This value will gradually increase while the application being debugged is minimized.
When the application is restored from minimization, the increased memory is released.

### Minimal reproduction project (MRP)

[mleak_vp_transparent_bg.zip](https://github.com/godotengine/godot/files/14809551/mleak_vp_transparent_bg.zip)

**Comment by Sauermann** [comment]
I have tried the MRP on Windows 10 with v4.3.dev.custom_build [29b3d9e9e] and there seems to be no memory leak.

![image](https://github.com/godotengine/godot/assets/6299227/682120de-95aa-48d9-b5ac-54e898f3c72b)

**Comment by clayjohn** [comment]
https://github.com/godotengine/godot/pull/90055 will fix this MRP. But the issue should still be reproducible with the following code:

```
extends Node2D

var v: SubViewport

func _ready():
    v = SubViewport.new()
    
func _process(_delta):
    v.transparent_bg = !v.transparent_bg 
```

@MizunagiKB Can you test with the code I provided here and let me know if the leak looks the same?

**Comment by MizunagiKB** [comment]
@clayjohn I have executed a build applying the specified pull request #90055.

Upon applying the pull request, the memory leak that I reported in my script no longer occurred.

![fig1_ss](https://github.com/godotengine/godot/assets/5047878/050d586a-83c1-4b22-a513-549301ba32bb)

---

However, the same memory leak occurred in the provided script (the code that changes the transparent_bg value every frame).

![fig2_ss](https://github.com/godotengine/godot/assets/5047878/49f52e06-f6f9-4ad0-af3d-9269ea747f10)

In the screenshot, the renderer is set to Mobile, but the same result was observed with Forward+.
